lib/semantics/resolve-names.cpp

```diff
@@ -6051,6 +6063,53 @@ bool OmpAttributeVisitor::Pre(const parser::OpenMPLoopConstruct &x) {
   return true;
 }
 
+const parser::Name &OmpAttributeVisitor::GetLoopIndex(
+    const parser::DoConstruct &x) {
+  auto &loopControl{x.GetLoopControl().value()};
+  using Bounds = parser::LoopControl::Bounds;
+  const Bounds &bounds{std::get<Bounds>(loopControl.u)};
+  return bounds.name.thing;
+}
+
+void OmpAttributeVisitor::ResolveSeqLoopIndexInParallelOrTaskConstruct(
+    const parser::Name &iv) {
+  Symbol *symbol{nullptr};
+  auto targetIt{ompContext_.rbegin()};
+  for (auto it{ompContext_.rbegin()}; it != ompContext_.rend(); ++it) {
+    if (parallelSet.test(it->directive) ||
+        taskGeneratingSet.test(it->directive)) {
+      symbol = ResolveOmp(iv, Symbol::Flag::OmpPrivate, it->scope);
+      targetIt = it;
+      break;
+    }
+  }
+  targetIt++;
+  if (symbol) {
```

Comment:
The increment should be inside the `if` because that is where `targetIt` is used. And `targetIt` doesn't need to be initialized.